cli/src/main/kotlin/commands/ClearlyDefinedUploadCommand.kt

```diff
@@ -75,33 +76,44 @@ class ClearlyDefinedUploadCommand : CliktCommand(
 
     private val service by lazy { ClearlyDefinedService.create(server, OkHttpClientHelper.buildClient()) }
 
-    private fun getDefinitions(coordinates: Collection<String>): Map<String, ClearlyDefinedService.Defined> {
-        val call = service.getDefinitions(coordinates)
-
+    private fun <T> executeApiCall(call: Call<T>): Result<T> {
         log.debug {
             val request = call.request()
             "Going to execute API call at ${request.url} with body:\n${request.body?.string()}"
         }
 
         val response = call.execute()
-        val responseCode = response.code()
 
-        if (responseCode == HttpURLConnection.HTTP_OK) {
-            response.body()?.let { definitions ->
-                return definitions
-            } ?: log.warn { "The REST API call succeeded but no response body was returned." }
+        if (response.isSuccessful) {
+            // Note: Cannot lift out "return" here as Result cannot be used as a left operand of "?:".
```

Comment:
Or work around this by restructuring the code a bit?

```
val body = response.body()
val errorBody = response.errorBody()

return when {
    response.isSuccessful -> when (body) {
        null -> Result.failure(IOException("The REST API call succeeded but no response body was returned."))
        else -> Result.success(body)
    }
    else -> when (errorBody) {
        null -> Result.failure(IOException("The REST API call failed with code ${response.code()}."))
        else -> {
            val errorResponse = jsonMapper.readValue<ErrorResponse>(body.string())
            val innerError = errorResponse.error.innererror

            log.debug { innerError.stack }

            Result.failure(IOException("The REST API call failed with: ${innerError.message}"))
        }
    }
}
```

Reply:
Good idea, done.

Reply:
You have done it in the wrong commit.

Reply:
Thanks for noticing, done.